matchers.go

```diff
@@ -57,6 +57,11 @@ func HTTP1Fast(extMethods ...string) Matcher {
 	return PrefixMatcher(append(defaultHTTPMethods, extMethods...)...)
 }
 
+// TLS matches HTTPS requests.
+func TLS() Matcher {
```

Comment:
Yes, I think ```cmux.TLS()``` should mean all version.
```cmux.TLS(tls.VersionTLS10, tls.VersionTLS11)``` matches only that specific version.

As for checking, yes, simply check the first byte is 22, it has the right version, and it's a handshake packet. 

Thanks!